extensions/amp-apester-media/0.1/monetization/companion/bottomAd.js

```diff
@@ -0,0 +1,74 @@
+/**
+* Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*      http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS-IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+import {Services} from '../../../../../src/services';
+import {createElementWithAttributes} from '../../../../../src/dom';
+import {dict, getValueForExpr} from '../../../../../src/core/types/object';
+const ALLOWED_AD_PROVIDER = 'gpt';
+
+/**
+ * @param {!JsonObject} media
+ * @param {!AmpElement} apesterElement
+ */
+export function handleCompanionBottomAd(media, apesterElement) {
+  const bottomAdOptions = getValueForExpr(
+    /**@type {!JsonObject}*/ (media),
+    'campaignData.bottomAdOptions'
+  );
+  if (!bottomAdOptions) {
+    return;
+  }
+  const enabledBottomAd = getValueForExpr(
+    /**@type {!JsonObject}*/ (bottomAdOptions),
+    'enabled'
+  );
+  if (
+    enabledBottomAd &&
+    bottomAdOptions['videoPlayer'] === ALLOWED_AD_PROVIDER
+  ) {
+    const slot = bottomAdOptions['tag'];
+    const bannerSizes = [[300, 50]];
+    constructCompanionBottomAd(slot, bannerSizes, apesterElement);
+  }
+}
+
+/**
+ * @param {string} slot
+ * @param {Array} bannerSizes
+ * @param {!AmpElement} apesterElement
+ * @return {!Element}
+ */
+function constructCompanionBottomAd(slot, bannerSizes, apesterElement) {
+  const width = bannerSizes[0][0];
+  const height = bannerSizes[0][1];
+  const ampAd = createElementWithAttributes(
+    /** @type {!Document} */ (apesterElement.ownerDocument),
+    'amp-ad',
```

Comment:
Do we ensure the `amp-ad` script is installed on the document elsewhere? If not, please do so.

Reply:
Is this intended to be a bottom sticky ad? If so, you should create an element
`<amp-ad sticky=bottom`

Reply:
@powerivq When we use 'sticky=bottom' ad block is sticky to the bottom of the window. It's not good for our case. We want to place ad block to the bottom of the media, but without CSS "position: fixed", which is added by sticky attribute.

Reply:
@caroqliu It's not clear why we need to check amp-ad script is installed. It's a "managed resource" (whether and when it will be loaded and displayed to the user is decided by the AMP runtime).

Reply:
Here is an example:
https://github.com/ampproject/amphtml/blob/9bb603c5803ec809c458ec54ca4cd20f12463a76/extensions/amp-auto-ads/0.1/amp-auto-ads.js#L52-L56